2

```diff
@@ -13,6 +13,10 @@ done
 
 [ -x /etc/rc.local ] && /etc/rc.local
 
+for script in /etc/rc.local.d/??-*.sh; do
```

Comment:
Why require this form? Just match any file in the script directory and execute if it's +x.